app/src/main/java/com/jalasoft/routesapp/ui/auth/login/viewModel/LoginViewModel.kt

```diff
@@ -0,0 +1,155 @@
+package com.jalasoft.routesapp.ui.auth.login.viewModel
+
+import android.annotation.SuppressLint
+import android.app.AlertDialog
+import android.content.Context
+import android.util.Log
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import com.google.firebase.auth.AuthCredential
+import com.google.firebase.auth.FirebaseAuth
+import com.jalasoft.routesapp.R
+import com.jalasoft.routesapp.data.model.remote.User
+import com.jalasoft.routesapp.data.remote.interfaces.IUserManager
+import com.jalasoft.routesapp.data.remote.managers.AuthFirebaseManager
+import com.jalasoft.routesapp.data.remote.managers.UserManager
+import com.jalasoft.routesapp.util.helpers.UserTypeLogin
+import dagger.hilt.android.lifecycle.HiltViewModel
+import javax.inject.Inject
+
+@HiltViewModel
+class LoginViewModel
+@Inject
+constructor(private val userManager: UserManager, private val auth: AuthFirebaseManager) : ViewModel(), IUserManager {
+
+    val loginIsSuccessful: MutableLiveData<Boolean> by lazy {
+        MutableLiveData<Boolean>()
+    }
+    val errorMessage: MutableLiveData<String> by lazy {
+        MutableLiveData<String>()
+    }
+    val signInGoogleOrFacebook: MutableLiveData<Boolean> by lazy {
+        MutableLiveData<Boolean>()
+    }
+
+    @SuppressLint("StaticFieldLeak")
+    var context: Context? = null
+
+    /*fun loginUserAuth(email: String, password: String): Boolean {
+        if (validateFields(email, password)) {
+            AuthFirebaseManager.loginUserAuth(email, password, {
+               // loginUser(email, password)
+                // change to true
+                loginIsSuccessful = true
+            }, { error ->
+                loginIs465564Successful = false
+                errorMessage.value = error
+            })
+        }*/
+
+    fun loginUserAuth(email: String, password: String) {
+        val valid = validateFields(email, password)
+        var user = FirebaseAuth.getInstance().currentUser
+        if (user == null) {
+            if (valid.isEmpty()) {
+                // if (validateEmail(email)) {
+                auth.loginUserAuth(email, password, {
+                    loginIsSuccessful.value = true
+                    Log.d("sda", "succ")
+                    print("success")
+                }, {
+                    loginIsSuccessful.value = true
+                    print("error")
+                    Log.d("sda", "${it}estrellita")
+                })
+                // } else {
+                //    loginIsSuccessful.value = false
+                //    print("fail")
+                // }
+            }
+        }
+    }
+
+    private fun showErrorAlert() {
+        val builder = AlertDialog.Builder(context)

```

Comment:
AlertDialogs should not be built on the viewmodel. This is UI, so it should be done on the Fragment or activity that this viewmodel belongs.

Reply:
This is from Adriana's code